ath79: add support for DPTechnics DPT-Module v1

daanpape opened the pull request:
Add support for the DPTechnics DPT-Module v1 SoM

Hardware
--------
* SoC:    Qualcomm Atheros AR9331-AL3A
* RAM:    64MB DDR2 (SK Hynix H5PS5162GFR-Y5C)
* FLASH:  16MB SPI-NOR (Winbond W25Q128FVSG)
* UART:   19200 8n1
* USB:	1 single USB 2.0 host port
* LED:
  * 1x system-LED on board
  * 1x WAN LED connection
  * 1x LAN LED connection

MAC Address:
```
 use        address(sample 1)   source
 label      80:20:e1:xx:xx:xx   dpt-factory@7c00
 lan        80:20:e1:xx:xx:54   $label - 1
 wan        80:20:e1:xx:xx:56   $label + 1
 WiFi4_2G   80:20:e1:xx:xx:55   $label
```

Installation from bootloader
------------

1. Connect to the serial console. Power up the device and interrupt autoboot when prompted or make GPIO11 high to interrupt boot.

2. Connect a host to any ethernet port and give it static ip 192.168.1.2/24

3. Enter the `httpd` command in uboot

4. Browse to the webinterface on 192.168.1.1 for uploading the compiled firmware

5. The bootloader will install the new firmware after uploading and the SoM will reboot.

Signed-off-by: Daan Pape <daan@dptechnics.com>

daanpape committed:
ath79: add support for DPTechnics DPT-Module v1

Add support for the DPTechnics DPT-Module v1 SoM

Hardware
--------
SoC:    Qualcomm Atheros AR9331-AL3A
RAM:    64MB DDR2 (SK Hynix H5PS5162GFR-Y5C)
FLASH:  16MB SPI-NOR (Winbond W25Q128FVSG)
UART:   19200 8n1
USB:	1 single USB 2.0 host port
LED:    1x system-LED on board
		1x WAN LED connection
		1x LAN LED connection

MAC Address:
 use        address(sample 1)   source
 label      80:20:e1:xx:xx:xx   dpt-factory@7c00
 lan        80:20:e1:xx:xx:54   $label - 1
 wan        80:20:e1:xx:xx:56   $label + 1
 WiFi4_2G   80:20:e1:xx:xx:55   $label

Installation from bootloader
------------

1. Connect to the serial console. Power up the device and interrupt
   autoboot when prompted or make GPIO11 high to interrupt boot.

2. Connect a host to any ethernet port and give it static ip
   192.168.1.2/24

3. Enter the `httpd` command in uboot

4. Browse to the webinterface on 192.168.1.1 for uploading the compiled
   firmware

5. The bootloader will install the new firmware after uploading and the
   SoM will reboot.

Signed-off-by: Daan Pape <daan@dptechnics.com>

hauke commented on target/linux/ath79/generic/config-default:
```diff
@@ -14,6 +14,7 @@ CONFIG_LEDS_RESET=y
 CONFIG_MARVELL_PHY=y
 CONFIG_MICREL_PHY=y
 CONFIG_MTD_REDBOOT_PARTS=y
+CONFIG_MTD_SPI_NOR_USE_VARIABLE_ERASE=y
```
Is `CONFIG_MTD_SPI_NOR_USE_VARIABLE_ERASE` really needed for this device? 
I think changing it will cause problem with other devices.

daanpape replied:
It is required because the dpt-persistent partition must be writable from within OpenWRT. Therefore a block size of 32kB instead of 64kB is required. If it can be set device specific this would also solve it

daanpape committed:
ath79: Removed superfluous USB port description 

According to instructions from @DragonBluep the DTS file is updated to not contain the USB port description as this is no longer required. There is no USB LED trigger so this is no problem. 

Signed-off-by: Daan Pape <daan@dptechnics.com>

Signed-off-by: Daan Pape <daanpape@users.noreply.github.com>

daanpape commented on target/linux/ath79/dts/ar9331_dptechnics_dpt-module-v1.dts:
```diff
@@ -0,0 +1,176 @@
+
+	port@1 {
+		#address-cells = <1>;
+		#size-cells = <0>;
```
@DragonBluep thank you, this is now updated.
